src/types/createRouteOptions.ts

```diff
@@ -22,63 +22,68 @@ type WithHooks = {
   onAfterRouteLeave?: MaybeArray<AfterRouteHook>,
 }
 
-type WithSingleComponent = {
+export type WithHost<THost extends string | Host = string | Host> = {
   /**
-   * A Vue component, which can be either synchronous or asynchronous components.
+   * Host part of URL.
    */
-  component: Component,
+  host: THost,
 }
 
-type WithComponentRecord = {
-  /**
-   * Multiple components for named views, which can be either synchronous or asynchronous components.
-   */
-  components: Record<string, Component>,
+export function isWithHost(options: CreateRouteOptions): options is CreateRouteOptions & WithHost {
+  return 'host' in options && Boolean(options.host)
+}
+
+export type WithoutHost = {
+  host?: never,
 }
 
-type WithComponent = Partial<WithSingleComponent | WithComponentRecord>
+export type WithParent<TParent extends Route = Route> = {
+  parent: TParent,
+}
 
-export function isRouteWithParent(options: CreateRouteOptions): options is CreateRouteOptionsWithParent<Route> {
-  return 'parent' in options
+export function isWithParent<T extends Record<string, unknown>>(options: T): options is T & WithParent {
+  return 'parent' in options && Boolean(options.parent)
 }
 
-export function isRouteWithComponent(value: CreateRouteOptions): value is CreateRouteOptions & WithSingleComponent {
-  return 'component' in value
+export type WithoutParent = {
+  parent?: never,
+}
+
+export type WithComponent<
+  TComponent extends Component | undefined = Component | undefined
+> = {
+  /**
+   * A Vue component, which can be either synchronous or asynchronous components.
+   */
+  component?: TComponent,
 }
 
-export function isRouteWithComponents(value: CreateRouteOptions): value is CreateRouteOptions & WithComponentRecord {
-  return 'components' in value
+export function isWithComponent(options: CreateRouteOptions): options is CreateRouteOptions & { component: Component } {
+  return 'component' in options && Boolean(options.component)
```

Comment:
notice I added `Boolean` checks in these guards. Just a little strictor to make sure there's actually a value